### ISSUE TITLE
Clarify intended use of the Vision document

### ISSUE THREAD

**Issue by michaelchampion** [w3c/AB-public#53]
Discussion surrounding a now-closed issue in https://github.com/w3ctag/ethical-web-principles/issues suggests that the Vision document more explicitly state how it can be applied to W3C's opperations. Presumably some understanding  of W3C's Values/Vision drives WG decisions, horizontal review, AC review of charters and PRs, and formal objections and their resolution. That's implicit in W3C practice today, but isn't spelled out AFAIK. 

When there was an engaged Director, Sir Tim Berners-Lee was the ultimate definer and applier of values/vision for W3C's work. Without his engagement going forward, It would be useful to write down the Values/Vision that the Team, AC, FO Councils, etc. should consider authoritative. I had always assumed that was the purpose of this document, but I don't see it stated anywhere.

I don't have draft language to propose, but some questions: 
- Am I missing something in the document stating how the Values/Vision are supposed to be applied in practice? 
- Do others agree this document's purpose is to define the core set of principles the W3C community SHOULD apply in reviewing proposed charters and standards? 
- Would it be useful to strongly suggest that team decisions, formal objections, FO resolutions, etc. be explicitly justified (at least when challenged) with respect to how they promote the values and vision outlined in this document once it is ratified?
- Do other foundational documents such as the Process, Bylaws, Member Agreement, etc. need to be modified to define or reference an authoritative Vision/Values statement, or can this document suffice to guide consensus-building?

**Comment by mnot** [comment]
Thanks for opening this issue, Michael, I was about to. This is a difficult topic, but I'd encourage the AB to listen carefully to Robin's concerns. I also suspect that if we can clearly articulate what the intended use of the document is, it might be easier to get broader participation.

To extend the issue just a bit -- once we understand the intended use, we will then need to assure that how it was created is appropriate to that use.

Saying that an AB document represents consensus isn't workable -- the most it can do is represent what the AB thinks. While the AB has consulted across many folks, the decisionmaking process is firmly in the hands of the AB; there's no mechanism to appeal decisions about it (and if there were, I suspect Robin would avail himself of it).

As a result, the document is effectively capturing what a set of long-time "insider" W3C participants think, not a community consensus. Yes, the AB could take it to the AC for ratification. That's not a substitute for a legitimate consensus process _during_ the document's formation.

I can see two alternative paths forward for this work:

1. The Vision document is explicitly an AB document that reflects what the AB thinks, and thus is only advisory in nature; it has no normative impact on how decisions are made at the W3C.
2. The Vision document is re-scoped to a consensus document, ideally homed in a WG. *Not* a CG, Task Force, or other informal body. It is explicitly scoped to be binding on decisionmaking (and one of the complexities is how that happens, of course).

Personally, I think we need #2, and I think we need it yesterday.

**Comment by cwilso** [comment]
That AB document *does* represent consensus of the AB - painfully so - and much of what you're seeing is a reaction to the suggestion that it is not a worthwhile effort, and should simply be thrown out and replaced by one person's work.

If you want to propose a Working Group, be my guest.  I would point out that it isn't "re-scoping the Vision document", it would be "establish a Working Group to build consensus from the ground up on the Vision of the W3C, the principles by which it should operate, and some rule of rules to be binding on decisionmaking" (which sounds painfully like it would have to be embedded in the Process as well) - and it would need to start with a blank slate.  I will put it mildly - I have concerns about the productivity of such a group, and I think it would be a mistake to start over.  Getting real engagement and real work on building consensus, not just writing text, has been the hardest part of the AB Vision effort thus far.

I would say that I would expect the same credence paid to the AB's work as the TAG's Ethical Web Principles and Design Principles - not binding, perhaps, but pretty strong guidance.  (And if you believe the EWP is binding, explain why, as it is not "consensus" either.)  To Mike's point, I think these documents need to be pulled together so that we at least have a guide; making anything binding on decision-making at this point in the W3C's evolution is going to be a gargantuan effort.

I'm concerned by the characterization that the Vision is "what a set of long-time "insider" W3C participants think, not a community consensus" and "there is no mechanism to appeal decisions about [the Vision document]" - because while technically that's true (the AB has not even published this as a Note), we have not been ignoring feedback and input.  Robin offered input, responded once, and then went silent on his own principle.  You make it sound like Robin has been ignored systematically - this is not the case.  We've been trying to agree on basic principles prior to detailing how we expect the organization to enforce those principles.

I agree that the Vision needs to be turned into actionable tactics for the W3C.  I disagree that that step comes before even agreeing to the basic principles.

To answer Mike's questions: 

- Yes, I do believe this document's purpose is to define the core set of principles the W3C community SHOULD apply in reviewing proposed charters and standards, as well as the process of developing those standards.
- I believe it would be better, a la the TAG's EWP and Design Principles, to deliver something in the nearer term rather than boil the ocean of rewriting the Process, bylaws, member agreements, etc first.  I would point out that the Bylaws were not written by consensus, nor were the member agreements.

**Comment by darobin** [comment]
(Merging with https://github.com/w3ctag/ethical-web-principles/issues/54 to try to help make this saner.)

> Robin offered input, responded once, and then went silent on his own principle.

Chris, it would help a lot if you tried to present facts in a manner more conducive to making progress towards some common ground. The topic at hand is the general approach to how to do values & vision, which is specifically called out as a key motivating factor in the [PR](https://github.com/WebStandardsFuture/Vision/pull/37): "this PR seeks to do two things: 1) to experiment with a longer format" all the way to "I am eager to hear what your take on this is." I specifically broke out the PR in a separate file because it's an experiment, to make it easier for the AB to evaluate it as such.

The whole of the feedback from the whole of the AB on this is David's "I think we're going to have enormous fun taking grand statements of high principles, in a Vision document, and working out what they mean in detail and practice." Hey, I agree, and I like the sentiment, but can we agree that that's not a lot to go on?

Looking more specifically at the interactions on the issue over time, there was a first short discussion in April 2021 which I participated in and that ended with David talking about drafting something, which seems like a satisfactory direction to me. Then there's a flurry of further comments _fourteen months_ later in August 2022, and not about the issue at hand. Again, I'm not trying to blame the AB or anyone, we all get busy. I'll be the first to admit that I dropped the ball in August — life happened, I was distracted with interviewing for new jobs and death in the family, and I expect that life happens just as much to others. What I'm getting at here is that, however, I don't think that it's accurate to represent the current doc as supported by the intensive work of a vibrant community. I also don't think that it's accurate to represent that my input is being seriously considered when the entirety of the feedback is that it would be fun to do, and then going entirely dark on the question.

Again, to repeat the point because I would very much like to get past the wall of defensiveness here: no matter how much consensus there is _inside_ the AB, and no matter how painful that was to achieve (which I totally, totally believe and sympathise with), that does not mean that the document is supported by broad consensus in the community that gives it some kind of protected status. Revisiting the approach is legitimate. Does the doc have more consensus than a proposal I wrote yesterday? I would hope so? Is it useful to compare the consensus of 3-4 people here and 9-10 people there? I _really_ don't think so.

Mark rightly points at the issue of legitimacy. I think that's the absolutely core issue. It's a key part that my proposal tries to address by grounding our values in a process we already have, that has stood the test of time, that has been developed by a huge community — and seeing how we can use that as a platform to build more of that. It's entirely possible that my proposal isn't the right one or is a bad implementation even if it's the right one, but it's at least a constructive and I believe credible attempt to get at this.

I don't want to go twenty rounds discussing whether the AB should have processed my input this or that way; I only brought that up to explain that I don't think it's fair or justified to claim that the process has been diligent. Bygones, etc. I don't care.

But can we please, _please_ move to a constructive place where we agree that it's not hostile or disrespectful or insulting to think that the current approach needs rethinking in order to get a legitimate outcome? I'd be happy to put energy into helping corral public discussion, but that's going to be a lot less pleasant if the AB doesn't agree that we can make significant change along the way.

**Comment by michaelchampion** [comment]
@mnot wrote:

> - The Vision document is explicitly an AB document that reflects what the AB thinks, and thus is only advisory in nature; it has no normative impact on how decisions are made at the W3C.
> - The Vision document is re-scoped to a consensus document, ideally homed in a WG. Not a CG, Task Force, or other informal body. It is explicitly scoped to be binding on decisionmaking (and one of the complexities is how that happens, of course).
 
I'm retired from the Process CG and AB, but as I understand it the AB plans to propose elevating  the Vision document to W3C Statement Status  https://www.w3.org/2021/Process-20211102/#memo once it is stable, widely reviewed, FOs processed, and the AC approves. That's exactly what a WG would have to do.  So, the AB Vision would be as much of a "consensus document" as any Recommendation.   

I agree the document needs some work to be "scoped to be binding on decision making"

**Comment by frivoal** [comment]
> I'm retired from the Process CG and AB, but as I understand it the AB plans to propose elevating the Vision document to W3C Statement Status https://www.w3.org/2021/Process-20211102/#memo once it is stable, widely reviewed, FOs processed, and the AC approves. That's exactly what a WG would have to do. So, the AB Vision would be as much of a "consensus document" as any Recommendation.

Exactly.

The first step along that path would be to publish it as a Note, which we should do soon. Reasons it has not happened yet include:
* that process did not exist when the document was started
* We had that Legal Entity on our radar, and that kept us somewhat busy for a little while.
* We have this director-free thing on our radar, and that too kept us somewhat busy (and still does. It's about to wrap but, but not in the past yet).

**Comment by michaelchampion** [comment]
At the risk of irritating all parties to this discussion:
- I think Mark and Robin make some good points -- the Vision document needs a lot of work to give a clear vision of where W3C should go and more concrete guidance on what values should be promoted/protected in drafting and reviewing standards.
- But I disagree that some new process or venue would help. We should all double down on reviewing and proposing changes to the document in this repo and not bikeshed about legitimacy, etc.  The AB are the legitimate owners of this and let's not undermine them!
- That said -- and I know I've irritated AB people with this assertion in the past, sorry -- the AB could try harder to  encourage the broader community to engage.  I personally have felt somewhat disempowered as fairly strong "we need to do this!" language in early drafts I helped write got watered down to  be less "negative".  OK, my bad for just giving up rather than filing comments on  PRs and alternative PRs... but AB folks could do better too.
- The biggest takeaway I got from Robin's start-from-scratch proposal in another repo was this issue: The purpose of the Vision exercise is to define some core principles to guide actual spec and charter drafting / reviewing, not to write platitudes.   Not all W3C members are going to agree with the result, and some will formally object, and a few probably leave in protest the the FOs are overturned.  So be it.  Better to have a critical mass of people pursuing a concrete vision of a better web than hide disagreements under bandages to fester into formal objections and political battles forever.

**Comment by darobin** [comment]
I like your "irritate everyone" approach Mike, thanks for putting that together. (I'm not irritated though.) One thing I want to insist on and get out of the way: I did not "start from scratch in my own repo," I put together an illustration of what I think a more robust & usable approach would be because just describing it was clearly not getting across.

Just a few quick points:

* I don't have a strong perspective on who is the "legitimate owner" but I do believe that the legitimacy of the output document hinges entirely on being produced by a vibrant community process. I think that this is more than "try harder to encourage the broader community to engage." Community engagement leading to consensus should be the one and only metric of success. The active participants might end up being a small community, but we should insist that it represent broad and diverse interests, and that it isn't just the same old us, much as I like hanging out with you all. For that to work, **there needs to be a clear value proposition that shifting the direction of that document will produce real-world change in how the W3C and the Web work**. No one's going to show up if it doesn't feel like it can change things.
* I think that it would be a serious misstep to take the current document and ship it as a Statement-track Note. This would send the signal that this document is somehow _the_ de facto way in which vision/values are captured, which will be used as a reason to shut down alternatives (as we've seen). The document simply does not have the legitimacy supporting it to justify that status. There are groups in the broader governance community (like the Ostrom Workshop or Metagov) that I would like to hear from since this is supposed to be load-bearing in web governance. I'm somewhat confident that they won't have anything to say about the current version and we won't gain traction generating involvement.
* To belabour the point, this is not a reflection on document quality. Even if this were my most favourite reading ever, I would _still_ object to its publication. It just doesn't have the legitimacy that it needs to have. This isn't an internal doc; this is for the Web.
* I agree with Mark that this needs to be done in a formal venue that provides the typical process guarantees. Putting it on the Statement track definitely helps, but there's work that needs to happen before that matching what we'd expect from a group — like for instance a call for participation, a clear and welcoming venue for people to help with and discuss, chairs and editors, calls, etc. Just putting it in a repo and talking to the AC doesn't pass the bar; we'd expect more from a group working on something less important! Maybe it doesn't need to be a WG, but it needs to enforceably and legitimately quack like a WG.

I would like to encourage us all to focus on making the vision document unassailably legitimate. This will be even more irritating, but I think that requires:
* A charter, that calls lists goals and sets the process for the work. If it's not a WG it should be like one. Maybe a STMT-track work item hosted by the AB can pass muster, I'm open to hearing about option.
* A call for participation. I think that it should emanate jointly from the AB, TAG, and Board, and it should be supported by communication. A list of parties to reach out to so as to garner community participation (compare this [list we made for the privacy work](https://github.com/w3ctag/privacy-principles/issues/217)).
* A chair or chairs, and the paraphernalia we associate with work items. Likely a cadence of calls and meetings, perhaps not super frequent but real nonetheless (and for all the people involved, not just the AB).
* Perhaps even a workshop.

Is this work and pain? Yup. Is it irritating? Almost certainly. But _we can't just ship a vision document that is produced in a manner that contradicts what it's saying_. I realise that this offers nothing more than further blood, toil, tears, and sweat. I hope that we can be swift; I'm adamant that we can't be insular.

**Comment by darobin** [comment]
I totally agree that W3C needs to pivot; what I believe it needs to pivot to is governance. Governance (of automated human/computer systems at planetary scale) is the biggest unsolved problem in tech. The majority of our more pressing problems boil down to the fact that we've built a world in which might makes right, in which collective action is near impossible, etc.

The W3C needs to decide if it's just standardising the Web SDK, or if it own stewardship of the Web. If it's the former, then governance doesn't matter much, but we also don't need a document that says our SDK is "for all humankind." That's kinda weird for an SDK. I think we (and many in the the wider community) largely want the latter. I also think that a strong, credible story about governance for the Web — not just that we are pivoting to it but that we have a treasure trove of experience with how to go about it (as offered by horizontal review) is a powerful hook to get funding.

Setting up the Board, updating the Process on a cadence, going Director-free — all these things are headed in the same direction. This doc should be part of that.

But we can't pivot to the thing without doing the thing. I can be convinced that we might not have to do all the things I listed, but not that a small coterie of insiders are legitimate in setting the vision for the web. We just aren't. I can't pretend that we are.

I'm repeating myself, but the key feature of the approach I have repeatedly advocated is that it relies on building from the massive successful investment in practical values that we already have as developed by the whole community. The specifics of the text don't matter; what matters in that approach is that we can establish the most lightweight frame possible with which to enshrine the outstanding work of the community, work that is supported by extensive consensus and that impacts real standards work every day. And having done that, we can use the existing process to keep iterating.

**Comment by michaelchampion** [comment]
> The W3C needs to decide if it's just standardising the Web SDK, or if it own stewardship of the Web. ... I think we (and many in the the wider community) largely want the latter

I think we've reached the heart of the matter here:  I originally saw  _agreement in the community_ as the first step toward W3C becoming an effective steward of the web, and the Vision is the vehicle for getting internal consensus to make that pivot.  You apparently think there is much agreement already on the stewardship mission and it's time to get on with the specifics of a governance system.

The TAG thread discussion and your proposal that spun out of it did convince me that the Vision needs to be less abstract and exhortative, and offer concrete, practical guidance to steer charters and PR transition reviews toward stewardship.  Yes, we can do a better job of building a critical mass of support for a stewardship mission by being specific about what the values we are promoting and how W3C standards work can help.

> Governance (of automated human/computer systems at planetary scale) is the biggest unsolved problem in tech. The majority of our more pressing problems boil down to the fact that we've built a world in which might makes right, in which collective action is near impossible, etc.

Woo... I think the best we can do for now is to harness W3C's collective brainpower and connections to actual product and policy makers to wrestle with the larger (possibly unsolveable) problem while doing what we can to nudge the web in a better direction. But again, that starts by getting a critical mass of the web/W3C community to accept the negative consequences of the web and the need for a stewardship mission.  From what I can tell from this repo, e.g. https://github.com/w3c/AB-public/issues/22 and https://github.com/w3c/AB-public/issues/14 which continues https://github.com/WebStandardsFuture/Vision/issues/12 ), that is still controversial.

I hope we can close those issues (and this one) with an acknowledgement of the web's serious problems and a consensus to pivot W3C's mission toward addressing them.    In other words, to craft a Vision that is NOT a "document that says our SDK is for all humankind"  but an acknowledgement W3C needs to move beyond 1990s techno-utopianism and adapt its culture and processes to both improve the web's technology and address its adverse consequences. The more specific and concrete the vision, values, and criteria for improvement are, the better the Vision will be.

**Comment by dwsinger** [comment]
I am actually fairly lost in what Robin and Mark see as problems here, and I think there may be misconceptions. 

For example, I'm not even sure whether the concerns are about the content of the document, or the way it's being developed.

If it's the content about the vision for the web, which of the following would more closely capture the concern?
1. A Vision for the web that doubles down on values is the wrong vision, wrong direction.
2. Values-based is the right direction, but these are not the right values.
3. These values are fine, but there are more that should be included.
4. This set of values is mostly fine, but the next step is missing, how they are connected to changes in our processes that will help us use these values to guide our operation and decisions.

If it's that the document mixes a vision for the web with a vision for the w3c, that's already been noted, and also that the two do have overlap or intersection.

If it's about the process of development, I think there is a fundamental misconception here and I don't know how it arose.  Mark says, above "Saying that an AB document represents consensus isn't workable -- the most it can do is represent what the AB thinks." There's an assumption in here that it is an AB-owned document. On the contrary, the AB (among other things)
* recognized that we no longer had an engaged Director whose visions and principles we could rely on to come into play when needed
* saw that our existing vision "lead the web to its full potential" is tired, short, and doesn't help us discriminate, make hard decisions
* realized that the W3C would need external support, and we'd need to explain to potential supporters why w3c and our future work matter.
* saw that if as representatives of the members it was for the AB to catalyze, jumps-start, community work here.

So, the AB listened (e.g., on the values point, to an ex-AB member, Mike here), and took the initiative, got text written, brought it repeatedly to the membership explicitly asking for input, supplied an editor, and have curated the process of consensus. That the AB took the initiative is not something to complain about, but applaud. It would not have happened otherwise.

Complaining that edits didn't simply get accepted side-steps the question of whether the proposed edits got consensus. After some years work, getting consensus that something is an improvement isn't always easy (I've had to work at it). The AB started this and sees this as a document the community needs. No, it's not "done", and the AB has not yet asked to get more formal community buy-in for this. I think the plan there is to make it into a Note and then take it through the Statement process to get consensus.

If the vision is fine as it is, but we have not yet done the next level of work, to work out how it becomes actionable, how our processes and actions will be modified to take it into account – make a suggestion. Please don't reject it because it's lacking something – supply that something. Likewise, I have a concern that we don't have enough 'specific features' or 'new things' in there, and that in some sense it's a 'mitigate harms' vision which could be seen as addressing a negative rather than proposing a positive; perhaps these could be addressed (e.g. by the TAG?).

In summary, I strongly agree that we, the W3C community, do need a new sense of vision for the future, and I think that the AB and the many contributors here have worked on it _for and with the community_, and we should be thanking them and helping.

**Comment by mnot** [comment]
Chris,

> That AB document does represent consensus of the AB - painfully so 

Of course. My point is that the AB's consensus -- even if ratified by an AC vote -- is not a great reflection of _community_ consensus, especially on a document that's so foundational. Yes, I understand that you've asked for feedback and consulted with various folks -- however, it's a document that reflects what the AB thinks about that, not one that the community has significant ownership of.

This happens sometimes in the IETF; the IAB comes up with a document purporting to reflect the consensus of the community, and are firmly reminded that there's a process for that and exactly one way to do it. However, they're more than welcome to write documents reflecting what the IAB feels (often after consulting with the broader community).

> I would say that I would expect the same credence paid to the AB's work as the TAG's Ethical Web Principles and Design Principles - not binding, perhaps, but pretty strong guidance. (And if you believe the EWP is binding, explain why, as it is not "consensus" either.) To Mike's point, I think these documents need to be pulled together so that we at least have a guide; making anything binding on decision-making at this point in the W3C's evolution is going to be a gargantuan effort.

So it sounds like your intention is path (1) above -- the document is advisory / persuasive, not binding, and it reflects AB consensus, not community consensus. I agree this is a pragmatic path forward to getting something out without a massive delay (and likely much gnashing of teeth), if we acknowledge its limits.

However, it's not clear that that's what's happening; Mike seems to have the impression that it's going to be a Statement, which is more like path (2). I think turning this document into a Statement at some point in the future after some community review process that's more than waving it by the AC might be a good step, but that's not going to happen anytime soon.

> I'm concerned by the characterization that the Vision is "what a set of long-time "insider" W3C participants think, not a community consensus" and "there is no mechanism to appeal decisions about [the Vision document]" - because while technically that's true (the AB has not even published this as a Note), we have not been ignoring feedback and input. 

There's a huge gulf between "we take feedback and input" and "we followed a process with broad stakeholder involvment, oversight, appeals mechanisms, transparency, and consequences for abuse of power." I would think that folks had seen enough of the former approach in recent times.

> Robin offered input, responded once, and then went silent on his own principle. You make it sound like Robin has been ignored systematically - this is not the case. We've been trying to agree on basic principles prior to detailing how we expect the organization to enforce those principles.

How Robin interacts with the AB is not my primary concern on this issue; I suspect the AB and Robin need to work that out separately. I'm concerned with what happens when this document is relied upon to support a contentious decision, and stakeholders come away feeling that it doesn't represent them, and that it couldn't have because of how it was created. 

Since Robin mentioned the Board -- this is _not_ wearing my Board hat; at this point I believe that while some aspects of this might have impact on the Team, they're operational.

**Comment by darobin** [comment]
@michaelchampion said:
> I think we've reached the heart of the matter here: I originally saw agreement in the community as the first step toward W3C becoming an effective steward of the web, and the Vision is the vehicle for getting internal consensus to make that pivot. You apparently think there is much agreement already on the stewardship mission and it's time to get on with the specifics of a governance system.

I like this framing, Mike, thanks. I would suggest taking it one step further: I don't think that we can produce or prove agreement on either the stewardship mission or the specifics of governance without driving that work in a manner that aligns with what the stewardship itself entails. Irrespective of the level of consensus in the AB, this has to come from and be made by the community.

@dwsinger wrote:
> For example, I'm not even sure whether the concerns are about the content of the document, or the way it's being developed.

A key point is that these two cannot be separated, or at least cannot be separated given the intended values. One way to understand the double concern is in terms of input and output legitimacy:

* Input legitimacy would be about how the doc is produced, eg. a broad consensus of very different constituencies or mostly just a small number of folks like us with tremendous access. The doc is much more of the latter.
* Output legitimacy would be about how the doc can work in the trenches in that it can help navigate contentious discussions (not answer them of course, but support them). To take a real example: is a consent popup enough to guarantee privacy? The doc doesn't help with that.

My suggestion — and again, it is only a suggestion, which I've been trying to explain in several ways — is that we can make a different trade-off. Instead of starting from scratch and listing all the values we could aspire to, we start from what we already have: a foundation of detailed, concrete values documents that already enjoys both kinds of legitimacy, a tried and true method to enforce them, and a process to produce more. I suggest that we then give that existing foundation a lightweight frame to enshrine it as such, and possibly list the areas that it doesn't cover which we would like to see some work on (and this could use the AB's doc as input).

**Comment by michaelchampion** [comment]
@darobin said:
> we start from what we already have: a foundation of detailed, concrete values documents that already enjoys both kinds of legitimacy, a tried and true method to enforce them, and a process to produce more.

As I understand the situation, there is no "foundation of detailed, concrete values documents" at W3C.  W3C was guided by Tim Berners-Lee's values and vision for a worldwide web in the early years.  It wasn't written down AFAIK,  but the Team and other participants developed a reasonably clear shared understanding of the fundamental values as they applied to web technology.

The problem is, the web's technology and business models have evolved rapidly, as Tim disengaged.  The AB, as I understand it, is trying to make the shared values more explicit, and (one can hope) specific enough to guide tradeoffs among between traditional values like "free", "private", "secure" and emerging values such as "sustainable". 

I see the AB leading a community effort to do more or less what you are suggesting.  Have they missed some concrete values documents to normatively reference?  Have they missed some values they should reference?  Have they invented some that aren't real W3C community values?  If so, this is an open GitHub repo, anyone in the "community" can file issues or PRs.

**Comment by michaelchampion** [comment]
@mnot wrote:
>This happens sometimes in the IETF; the IAB comes up with a document purporting to reflect the consensus of the community, and are firmly reminded that there's a process for that and exactly one way to do it.

Could you point us to the IETF community's "exactly one way to do it" community consensus process? And maybe examples of it working well for a "human" matter  like values / vision as opposed to a "technical" protocol standard?

**Comment by dwsinger** [comment]
(Guessing here) the IETF uses RFCs for a lot of things; and moving a document from I-D to RFC is well known.

We have group drafts (working documents), group Notes (group consensus) and W3C statements (W3C Consensus). I think we are or are planning to follow that route.

**Comment by darobin** [comment]
@michaelchampion wrote:
> As I understand the situation, there is no "foundation of detailed, concrete values documents" at W3C. W3C was guided by Tim Berners-Lee's values and vision for a worldwide web in the early years. It wasn't written down AFAIK, but the Team and other participants developed a reasonably clear shared understanding of the fundamental values as they applied to web technology.

On this point I disagree. We have a _wealth_ of documents in our horizontal review activities telling us in very actionable detail what it means that `There is one interoperable world-wide Web` (TAG), what it is to `be safe for its users` (TAG, WebAppSec, PING), what `for all humanity` looks like (I18N, A11Y). These do exactly the work that has repeatedly been mentioned as what we expect from the vision doc: they provide extremely valuable support for contentious discussions and they actively guide where the Web is going next, at least for the areas they cover. They generally benefit from a high degree of legitimacy. They are also enforced, which is the only thing that makes values anything other than words. And we have a process to make more. Are they perfect? No. Is it messy? Yes. Do they cover everything we'd like to cover? Not nearly enough. But they exist!

That's why I'm advocating a different path:
1. Recognise the treasure trove we have — warts and all — by elevating horizontal review to "how we do values".
2. Provide a thin and lightweight frame around it to make it the official position of W3C that this is how we do values, essentially ratifying the de facto practice. The more limited we can make this document, the easier it will be to make it legitimate.
3. Guide some cleanup (to avoid having bits and bobs of documents that don't look the same scattered around) and perform some gap analysis. This could build from the current AB's doc.

Is this perfect? No, and it's just a suggestion. But it does have some benefits:
1. For the areas that are covered, it provides a level of legitimacy that no other process we have can hope to match.
2. It enshrines one of the most load-bearing parts of W3C. It recognises the work of the community and makes it central and foundational to how we work — _as it should be_.
3. For the areas that aren't covered, it tells us how to plug them one by one over time.
4. It's enforceable, in fact it's already enforced. This means it's actual values and not just exhortations.
5. It's something that I would feel a _lot_ of confidence going to funders with. "Do you like what we did with {accessibility, internationalisation, keep 5 billion people safe}? We have a clear and established way to reproduce that for $x$." We're leaning into experience, community, established process.

**Comment by cwilso** [comment]
Since the TAG is an elected body, same as the AB, how is any document they produce more "consensus" validated than what the AB could produce?

**Comment by michaelchampion** [comment]
Agree that values in the Vision document should align with current horizontal review areas.  Are there any not mentioned in the Vision?   

The  more interesting question is how new horizontal review criteria get added.  CGs can *incubate* criteria that could be used in wide review.  But who decides, for example whether "sustainability" is a core W3C value that MUST be respected in Recommendations?  Having a Statement referencing that Value approved by the AC (and that survives the formal objection process) seems appropriate.

**Comment by cwilso** [comment]
And that, Mike, was precisely why we wrote the Vision starting where we did; because we needed to agree on WHAT the values and principles were, before we detailed how they would be enforced.  Indeed, we specifically focused on privacy, security, internationalization, equity, accessibility, and more.  Those things DO align with HR groups, on purpose; and the strategic structure of that is the next step.

**Comment by darobin** [comment]
@cwilso asked:
> Since the TAG is an elected body, same as the AB, how is any document they produce more "consensus" validated than what the AB could produce?

How the group is convened is only a small part of the legitimacy of its output. The TAG's work on, say, XML Versioning has very little legitimacy because it's just the output of a few TAG people, bright as they may have been, and while they spent an inordinate amount of time and energy on it and presented it to the AC, it never garnered broader support. 

Conversely, when the TAG publishes the Web Platform Design Principles, these have been through a veritable crucible of consensus-building. The PoC has been used in practice over and over again, etc. I do not doubt that the AB _can_ also produce this kind of battle-tested document; I am pointing out that we have documents that are load-bearing of our values and already exist as battle-tested, enforced, staffed, etc. It strikes me as a good idea to build on that.

**Comment by mnot** [comment]
> This document was published by the [Technical Architecture Group](https://www.w3.org/groups/other/tag) as a Group Note using the [Note track](https://www.w3.org/2021/Process-20211102/#recs-and-notes).
>
> Group Notes are not endorsed by W3C nor its Members.

**Comment by cwilso** [comment]
@darobin you didn't answer my question, and did not describe how the TAG design principles are any more of a "crucible" than the AB's open vision discussion has been.

**Comment by darobin** [comment]
@michaelchampion wrote:
> Agree that values in the Vision document should align with current horizontal review areas. Are there any not mentioned in the Vision?

If there is an intention to align with HR areas, it's not written into the document. If we don't write it into the document, we hit the problem that we're ignoring very significant work in the community and arguably one of the W3C's (and the web's) strongest assets. It also might produce values that aren't aligned with what we actually enforce in those areas. If on the other hand we do write it into the document, then we hit the problem that some of the document maps clearly to a thick and solid foundation of enforced values with a strong consensus base, and some of it maps to nothing.

> The more interesting question is how new horizontal review criteria get added.

This is absolutely, absolutely key, and is a reason why I don't think that the "list values now, figure out how to implement them later" plan serves us well. If (just taking an example) we say that sustainability is a value but we don't stand up HR to define, implement, and enforce it then we're basically lying. I really don't like how that sets us up. Putting up an HR review area is hard work, both in the sense of being intensive and difficult. What happens if we then don't have a credible Sustainability HR after one year? After five years? After ten? Do we then keep it as a value and damage our credibility, or do we remove it from our values after a while because clearly we're not doing it and damage our moral standing?

That's why I propose to start from what we have rather than intend to connect with it later. It's also why the suggestion I made had as its change process that you only get to add a value if you can stand up the HR review process.

I, and I suspect all of us, would dearly like to have HR for sustainability and equity and a bunch of other things. But standing up a new HR is a significant undertaking and I wouldn't assume that we can get all of these up and running within a reasonable time frame. I think that promising that we have values without doing the work to make them real sets us up for failure.

I wonder if a way forward would be the split the approach? Have part of the v&v be "how W3C does values" and include the values that we can prove we have as well as the framework for managing and evolving them. And then a "what are we missing" document that carries out gap analysis and works towards a plan to progressively move values from one doc to the other?

**Comment by darobin** [comment]
@cwilso Then I'm afraid that I don't understand your question? I don't think (and don't see that I have ever said) that the TAG had more consensus (or legitimacy) than the AB, however some document have more consensus/legitimacy than others. I also didn't say that the TAG's WPDP were a crucible, only that they had gone through one that the AB vision doc hasn't. We can debate (and I would, but elsewhere) whether everything that is in WPDP should be, but most of that doc is the result of extensive discussion with a broad community and principles that have been used in practice to resolve contentious discussions.

**Comment by dwsinger** [comment]
I am not sure why we're having an either/or discussion here. The TAG's ethical web principles and web design principles apply and document the consequence of values. I read EWP to help with my contribution to the Vision. The TAG documents are valuable, indeed to the extent that I think they need to be elevated to Statement status. But they are not the short, punchy, straight-language statement of our values and vision that we also need and that the AB document is attempting to give birth to. The three should absolutely be congruent.

**Comment by cwilso** [comment]
Indeed, I agree, and yes, the EWP are explicitly referenced as inspiration and companion in the vision document.  I'm trying to understand why a very similar group, following if anything a less openly inviting process, is referred to as having gone through a consensus crucible, while the AB's effort to define a vision statement in the open, over multiple years, is not.

(If I'm correctly interpreting Mark's answer, it is that the TAG documents are not more definitive, since they are Notes.)

**Comment by michaelchampion** [comment]
@mnot wrote:
> Group Notes are not endorsed by W3C nor its Members.

and @cwilso wrote:

> (If I'm correctly interpreting Mark's answer, it is that the TAG documents are not more definitive, since they are Notes.)

At TPAC 2022 in Tess said in her lightning talk  "The Ethical Web Principles document is on [the Note track](https://www.w3.org/2021/Process-20211102/#note-track) and **we hope to elevate it to a [W3C Statement]**(https://www.w3.org/2021/Process-20211102/#statement).

So I believe the TAG and AB have similar plans for these documents: Publish as a Note to drive discussion and consensus building, then elevate to a Statement AC   to get the same sort of W3C-wide status Recommendations have.

**Comment by michaelchampion** [comment]
@darobin wrote:
>I think that promising that we have values without doing the work to make them real sets us up for failure.

Not sure anyone would disagree.  I don't see in the  current draft of the Vision document any references to values that don't have real horizontal review activities in place.  (I do agree it would help to explicitly reference them)

The main "value" mentioned that doesn't have a mechanism to make it real is "We put the needs of users first: above authors, publishers, implementers, paying W3C Members, or theoretical purity."  I hope nobody is suggesting the Vision *NOT* reference the principles in RFC 8890 (which is indirectly cited via a reference to the Ethical Web Principles which cites https://www.w3.org/TR/design-principles/#priority-of-constituencies which cites RFC 8890) .  I'm not sure how W3C can make this "real" (a Priority of Constituencies Review Board???? ...shudder) but I'd be comfortable with somewhat aspirational language in the Vision saying W3C will be working on how to put users first.

Likewise Sustainability.  It's not mentioned in the current Vision.  I wish it was mentioned,  perhaps with a reference to the CG incubating sustaina review criteria, and a caveat that it is not yet "real."  This would be useful as a "stake in the ground" -- we have concerns about the sustainability of the web,  and are working on making it part of the wide review process.  This gives  fair warning those whose tech might raise concerns (advocates for Proof of Work blockchain tokens perhaps) that W3C might not be a friendly environment for incubating / standardizing their work.

**Comment by darobin** [comment]
@cwilso wrote:
> Indeed, I agree, and yes, the EWP are explicitly referenced as inspiration and companion in the vision document. I'm trying to understand why a very similar group, following if anything a less openly inviting process, is referred to as having gone through a consensus crucible, while the AB's effort to define a vision statement in the open, over multiple years, is not.

You keep referring to this as a difference of legitimacy between groups when I keep pointing out that the difference that matters in this case is between documents. As I said (twice already), I don't believe that one of the TAG or the AB is more legitimate than the other. I have given examples of documents which I think are more legitimate than others, notably the WPDP. Note that this is *not* the same thing as the EWP.

About the EWP I have specifically said:
![ISSUE 2: It's not obvious that Ethical Web Principles actually succeeds according to the high bar for values & vision which this document sets out.](https://user-images.githubusercontent.com/38491/218659440-7db2b4d0-def5-41d9-b4fa-21ae91c39714.jpg) The content of the EWP is _somewhat_ more precise than the AB doc, but overall I find it still too abstract and lacking in applicability. 

So if your question is in fact "don't the EWP and the AB doc have the same problems?" then my answer is broadly "yeah, that's what I've been saying."

**Comment by chaals** [comment]
It seems @darobin  is suggesting that "how the values will be enforced" is a critical aspect of what they are. (Whether that is really his position or not, it makes sense to me). Which means that until that starts to get captured, it's hard to know if we're on the right track.

It also seems clear that he is suggesting the well-known Working Group process is the right one for dealing with key governance issues. I agree with that assertion. It was the mechanism deliberately chosen two decades ago to develop the Patent Policy, which was probably the most contentious thing W3C did until EME (also developed in a Working Group...). W3C's Patent Policy is a very clear statement of values, structured precisely around how they get enforced. It took a lot of discussion to create, but whether you view it as too restrictive or not restrictive enough it has been stable, accepted by the community, and applied in practice many times over the last two decades.

Documents developed by either the TAG or AB effectively have an "inner group", and in the case of the AB who (IMHO rightly) do not entirely in public view, this can (and in concrete cases, some trivial and some less so, does) lead to decisions being taken without transparency. Whether the decisions are right can be less important than the fact that from the outside looking in there is a disincentive to challenge such decisions. That disincentive comes from a perception that the body working _in camera_ has decided that particular decision is theirs to make, a perception that is real even if inaccurate.

(NB I wrote this 12 hours ago, and let it sit while I thought about it. During that time, 5 people wrote 10 comments. This reality alone makes it difficult to effectively participate in a process, which strengthens the suspicion of achieving "consensus by attrition" rather than real "buy-in". I think that's relevant to this particular discussion).

**Comment by dwsinger** [comment]
I wonder if "enforce" is the right word? I rather think that we expect that the community will use values to help guide approaches, and that in hard cases (e.g. formal objections) the Council should use values to help guide the decision.

**Comment by darobin** [comment]
@chaals said:
> It seems @darobin is suggesting that "how the values will be enforced" is a critical aspect of what they are.

Indeed, in fact I'm saying a stronger version of that: unenforced values are just words.

@dwsinger asks:
> I wonder if "enforce" is the right word?

It's a loaded word in that it tends to evoke what we would often think of as enforcement, typically the cops swoop in or the system punishes you. But there are plenty of different ways in which enforcement works that can be modulated based on the context. If people tut-tut you or give you death stares when you use your phone in the silent carriage, that's enforcement. If you know that the people you work with will think less of you and maybe won't be inviting you to drinks because of your norm violation, that's also enforcement. (We can use another word; the institutional analysis framework which I favour calls it the "Or Else" component which feels more ominous!)

This brings us straight to the question of what the appropriate and effective level and approach to enforcement we believe can work in any given case. If we expect any of this to guide approaches, then there has to at least be some way in which approaches that don't align with the values are made to feel that they are off course — some degree of enforcement. We have repeatedly made the mistake of just assuming that obviously everyone agrees about what "privacy" or "transparency" or whatever other value means, and that the definition spreads to newcomers via some kind of osmosis. That has repeatedly not worked: we need a reliable, repeatable way to make sure that this can't be missed or mistaken. If you have a strongly shared culture, you can enforce less (just rely on the fact that norm violations are frowned upon) but we keep tripping ourselves up when we assume that. By the time you put enough definition of these things and make sure that groups can't ignore them… you're basically right in HR territory.

Also, if the Council motivates decisions from this, they have to be detailed enough that it doesn't feel like any Council member can just make up a definition to match their ad hoc needs. Otherwise, that will seriously undermine the Council's authority.

**Comment by frivoal** [comment]
I think that how values get applied/enforced is important, but that how that gets done during any of:
* charter review
* spec development
* negotiation of a partner agreement
* recruitment of a CEO
* identification of appropriate / inappropriate targets for the bizdev team
* setting of a 5 year strategic plan for the organization
* …
is different, and so the mission / value statement cannot contain the enforcement rules for all contexts. It should be a document that gets referred to / instanciated in / used as a framework for various contexts, and each of them should explain how they are going to apply / enforce these for their particular contexts.

**Comment by darobin** [comment]
@frivoal One can always set an impossible bar and then observe that it isn't cleared :)

No system can give you a mechanical decision procedure for everything. One common way of handling this is that you have more or less mechanical enforcement where you can, and votes for everything else.

The HR system doesn't work well to evaluate "targets for the bizdev team", but then neither does the AB's doc. We keep assuming that we all have the same values in our heads, but I think we have a lot of evidence that we don't. There is enough room for interpretation that, based on the current doc, we could justify completely different decisions in evaluating bizdev targets.

I think that "we don't have clearly defined values, here is my considered opinion" is better than "my opinion is aligned with the W3C's values" when that can't clearly be argued. The latter decreases the credibility of the document and of the institution. You risk ending up with values that are like "freedom" in US politics.

**Comment by cwilso** [comment]
Please leave politics out of the discussion.

>We keep assuming that we all have the same values in our heads, but I think we have a lot of evidence that we don't. 

YES.  That is PRECISELY the point.  So we need to detail those values and what they mean, and we'd better agree with them at the core prior to attempting to detail how they will be "enforced" (or you will end up with "enforcement" that knowingly, willfully has loopholes).  

You're acting as if we think the Vision document is *done*, and either it completes its purpose or it doesn't.  That is absolutely not the case.  It has been difficult to make it this far because we HAVE been trying, among an elected representative set of Members, to try to capture what we believe the values should mean, and reach some kind of consensus.  We have lots more work to do.  This will never be perfect consensus, and all Members will NOT be made happy - witness the disagreement with the TAG's Ethical Web Principles, for example (https://github.com/w3ctag/ethical-web-principles/issues/66).  That doesn't mean those are the wrong principles.

@michaelchampion I think your question as to the purpose has been answered, though you may disagree.  I'd like to ask what you'd like to see in order to resolve this issue.

**Comment by mnot** [comment]
For this issue to be closed, I'd expect something to be reflected in the document -- one of the very bad habits we seem to have is expecting that everyone has all of this context...

**Comment by cwilso** [comment]
And that's why I was asking the issue filer what that "something" should be, in their opinion.  (Feel free to give your answer too, of course.)

**Comment by michaelchampion** [comment]
Yes, I think we have answered my original question and am happy to close the issue or suggest the AB close it (or merge with #22) .   @cwilso  proposes in https://github.com/w3c/AB-public/issues/22#issuecomment-1428816738 putting a high level description of W3C's  values /  principles in this document, and a more operational description in a later revision or different document.  I agree.

I also agree with @mnot that it would be best to close this issue with a reference to a PR or updated document with specific language saying the purpose is something like "Clearly define the principles and values that should underlie our operations."

 I would like to see some sort of "manifesto" about pivoting W3C's mission to stewardship of the web platform's integrity, it's clear there's little in current practice to refer to and probably not broad consensus this is what W3C is all about.   But that might have to be the mission of a "community" that transcends W3C (and WHATWG, TC39, IETF, etc) that needs to be founded on the basis of some guiding principles, not have them bolted on after 25 years.

**Comment by darobin** [comment]
So, I'll step away from continuing the blow-by-blow as that doesn't feel like a productive avenue. (Happy to take specific aspects offline, as always.)

I think that the consensus I'm hearing is:
* The current AB document will not have an operational impact and is not expected to.
* The document will be updated with a statement to capture that.
* A separate document will be developed to render the vision operational and provide a foundation for governance.

Is that a fair and accurate representation of the consensus?

If so, I don't have a strong opinion about whether this should be one document or two documents. I have a slight preference for one, but I can totally live with two. However, I think that both documents should be put forward in tandem. I don't believe that we can claim to have values that aren't reflected in concrete operations (and I don't believe that I have heard anyone here arguing that, so maybe that too is part of the consensus). I would very much like to ensure that we don't end up with a document that is difficult to endorse as a Statement because we don't understand its operational impact.

I realise that this is work (but then we all know that) and that it would be unfair to just dump extra work on the AB or the editors. Assuming we set up some sort of structure to carry out this work with a broader community, I would be happy to volunteer to help work on the second document.

**Comment by michaelchampion** [comment]
There are plenty of other issues that need review and proposals. For example https://github.com/w3c/AB-public/issues/34 … what widely accepted definitions of some of theses principles can we point to? “Privacy” rights are defined in various laws and regulations, for example, should they be cited? And  more explicit reference to RFC 8890 might help anchor the Users First principle that gets no explicit W3C horizontal review (yet)

**Comment by darobin** [comment]
I didn't intend it to be harsh, I'm only trying to be crisp. Maybe a better way to phrase it is how Chris put it in #22: the AB does not expect the Vision doc "to ensure the organization's work reflects [its] values and principles". I'm assuming that your comment is directed at the statement that it won't have operational effect and the idea is to soften that to say that it doesn't ensure operational impact?

Reformulating (trying to converge on a resolution):

* The current AB document is not intended to ensure that the organization's work reflects its values and principles.
* The document will be updated with a statement to capture that.
* A separate document will be developed to ensure that the organization's work reflects its values and principles.

Do this work better? The rest of my comment holds.

Regarding your other question @michaelchampion, I don't think that it would be a good idea for a foundational governance document of a transnational organisation to reference specific laws (possibly with the exception of global legal frameworks, in the cases in which they exist). For privacy, we do have the [Privacy Principles](https://w3ctag.github.io/privacy-principles/) that are shaping up. They haven't been beaten up by as much input as I'd like or edited for clarity enough, but they're grounded in research and we're [ramping up for wide review](https://github.com/w3ctag/privacy-principles/issues/217).

**Comment by dwsinger** [comment]
@darobin saith:

> - The current AB document is not intended to ensure that the organization's work reflects its values and principles.
> - The document will be updated with a statement to capture that.
> - A separate document will be developed to ensure that the organization's work reflects its values and principles.

I don't think it helps to describe it as an AB document, but maybe we need to if (and I hope they will) the AB puts it on Note and Statement track. I don't think a document can _ensure_ anything, but it can and should be used as a tool by members, team, and particularly when hard questions arise (e.g. FOs, but hard questions come up in other contexts).

A companion document could be created that explains:
* what is the intended development route and hoped-for final status of this; to AB note, to consensus statement
* that the current document is intended to provide a base of documentation of our values and principles.
* that by itself documentation does not ensure anything, but it can and should be used as a tool when needed (e.g. when resolving FOs).
* further work may be appropriate to develop documentation on using the values and principles to ensure that the organization's work reflects those values and principles.

**Comment by cwilso** [comment]
The current effort has been focused on gaining consensus on WHAT vision and principles we need to have as a path forward.  There was a general understanding (I won't claim a consensus, since I don't recall ever explicitly asking the AB that question, but certainly more than just one or two people agreed) that we needed to continue with how to apply those principles in practice, so I would not say it was a non-goal.  This might end up in a separate document, but I think it would be a mistake to plan for that at this point, since it seems the operational advice might easily deviate from the core principles.

It is definitely the plan to put this on the Note and Statement track soon.

I'm going to mark this issue as "proposed for closing", as it's effectively at this point a duplicate of #22 ; it sounds like there's a request for this to hang around open until there's a pull request for that issie.

**Comment by darobin** [comment]
@dwsinger: I am describing it as "the AB document" because for the time being that is what it is. Evidently I would hope that it is more than that if it comes to fruition.

@cwilso: It is evidently the AB's prerogative to close this issue. I think I have made the case and that putting further energy into trying to help improve this document will not be useful. I have literally tried to offer help every single way that I can think of and can only express sadness at the thinly veiled hostility with which it was received. If I played a part in triggering it, that was not my intent.

Before parting I will simply note my dissent. I personally do not believe that producing documents without the accompanying means to hold ourselves accountable to them is in line with the practices and values of the W3C, and doing so fails to further its mission. Barring meaningful progress on this issue I don't anticipate to be able to vote in favour of the resulting Statement.

**Comment by cwilso** [comment]
@darobin I believe that you have expressed your perspective that the current work is a dead end.  A number of others believe (and I would even venture to say "the consensus of the AB is", though of course I have not asked for a formal declaration of that lately) that this work is not a dead end.

I understand that you believe there should be a strong focus on enforcement/adherence to the principles.  I don't think you have heard me that I agree with you in this; where we disagree is that you appear to maintain that that enforcement must come first, before we even reach agreement on what the principles are.  I do not agree with this; however, as I've said several times, I believe the vision won't be useful if is simply stops at defining some aspirational principles; it needs to drive strategy and tactics in the W3C.  In my opinion, working out principles that can reach consensus on is a very hard part, and it needs to be done first.  Regardless, I would agree with you that producing principles without any means or thought to how we can hold ourselves accountable would make those principles much less useful; it was always the idea to build that progression and provide that detail.

I understand there is also additional input you would have on other principles, like the pull request you filed on origin sovereignty a while ago (#17).  I will say again - that work was not rejected; I don't think you had built consensus that it should be adopted as a principle, I expressed some concerns, and the PR itself was flawed in that it did not (probably purposefully did not) integrate with the rest of the Vision, but sat alongside as a separate document with no cross references.  All that stated - I did not ever consider that issue closed, but something we just hadn't gotten back to.  The Vision was effectively laid aside while we spent all our AB time working on bootstrapping the LE.  If you care about that principle, I would encourage you to go back and participate in the discussion there, and try to build some consensus - if you feel this work is inevitably doomed anyway, then okay, just close it.

As to "thinly veiled hostility", I would encourage you to think first about how you would react if you had been driving a collaborative effort with a dozen or so other people, publicly presenting your work several times and asking for public engagement, and were told by someone (who had not been saying so in the past few years you'd all been working on it) "this is a broken process, the process must start over, here is a document I wrote last night to replace the work you've all been working on."   That is dismissive of the work that the entire AB has been doing collaboratively over several years, effectively declaring it worthless; and I must admit that disrespect of work put in to building consensus puts me on the defensive.  I did not intend to be hostile, and I would much rather work together.

I am attempting to move this issue (and the Vision as a whole) forward with consensus of the AB at the very least, and ideally the Membership - certainly the bulk of the Membership.  I will note that it is highly unlikely that the entire Membership will agree with everything in the Vision, especially some of the Principles - in the same way that some of our Members today disagree with other principles expressed, say, in the TAG's Ethical Web Principles (e.g. Sustainability https://github.com/w3ctag/ethical-web-principles/issues/66).  That doesn't mean those principles are wrong, or that this Vision is.  In fact, I would even go so far as to say if the Vision doesn't garner some dissent, it is probably far too weak and not forward-thinking enough.

**Comment by chaals** [comment]
It is unfortunate that it's really easy to see two "sides of an argument" in this discussion - and more so that they seem to represent entrenching positions. Given the individuals are of good will, and are mature adults who can recognise the need to reset an re-engage in a discussion, I hope it doesn't continue this way.

I note that the issue began with

> Discussion surrounding a now-closed issue in https://github.com/w3ctag/ethical-web-principles/issues 

I am still unaware what that discussion was - although it seems there is an assumption that people know that. It's unfortunate that the mention was so vague.

It is not obvious to me that there has been a formal effort to reach consensus that, for example "The goal of this document is to define the core set of principles the W3C community SHOULD apply in reviewing proposed charters and standards" (or some other similar or even wildly different goal). I think that would be helpful, because I think we could reach agreement on something like that.

Mixed in with that discussion that I think is the core of this issue, there seems to be a question of "how to get there" - i.e. what is the right path to develop such a document. 

It seems we do not agree on that, and without an effort to reach some better agreement on how this might happen it seems entirely possible that the effort spent on this work will be wasted because we cannot get agreement that the outcome matches its purpose, or that a lot of effort will be wasted because people work very hard on a document, present it, and are told to specify the use case and problem first and then have all the discussions all over again in a different forum.

**Comment by chaals** [comment]
It also seems that this issue is a duplicate of #22. I would support, as part of resetting the framing and trying to do the discussion better, closing this as a duplicate and trying to have a less defensive discussion there.

**Comment by dwsinger** [comment]
I certainly don't think I have an entrenched or entrenching position, as that would imply that I understood something well enough to oppose it, and [I still don't](https://github.com/w3c/AB-public/issues/53#issuecomment-1427199627). Perhaps this comment from Robin helps me understand a little better:

> Before parting I will simply note my dissent. I personally do not believe that producing documents without the accompanying means to hold ourselves accountable to them is in line with the practices and values of the W3C, and doing so fails to further its mission. Barring meaningful progress on this issue I don't anticipate to be able to vote in favour of the resulting Statement.

I have two problems here. The first is the assumption that if we adopt consensus values, as a consortium (Statement), we'd want to but have no way to enforce them. I don't think that assumption is true today:
* once we have consensus values, they can be used by, and referred to by, anyone, particularly those writing charters or defining the architecture of a technology;
* they can be used as the basis for an appeal or formal objection, if anyone in the community feels that they have not been properly taken into account;
* they can be used as a tool in deciding appeals and formal objections.

The second aspect that puzzles me is the idea that somehow the order – first, write down your principles, and then consider what procedural changes might be needed so that they become embodied in work – is wrong, and we should do it the other way round. I can't imagine getting a good reaction if we were to introduce provisions designed to enforce values and principles, which were to be defined later. 

I _will_ continue to oppose including text that either people don't understand, or that hasn't got consensus, and I hope the editor will continue to do the same. That's normal and not unreasonable.

Finally, it seems to be way early in the process to be making threats (e.g. "Barring meaningful progress on this issue I don't anticipate to be able to vote in favour of the resulting Statement"), when we're at the stage of trying to understand each other and make improvements based on that understanding. Can we take a deep breath and engage in meaningful discourse, please?

**Comment by darobin** [comment]
@cwilso Thank you for taking the time to write down your thoughts in detail, I appreciate it. I do believe that on at least some important points, we have been talking past one another. I would like to see if we can't at least pare down the misalignment. (Thanks @dwsinger for adding some further thoughts as I was writing.)

I do not believe that "the current work is a dead end." I believe that its approach needs to change if it is to have an impact in the real world — or at least the level of impact that I would like to see it have. Making it impactful would require, in my opinion, significant edits to the document so as to define principles in a way that can be used in arguments (generally in support of deciding the things that @frivoal listed). I do believe that the _approach_ of listing principles but not defining them clearly enough that people can agree or disagree with them is a dead end yes (and maybe that's what we agree to disagree on). I _think_ that you might actually agree with that too, though, but that we have different expectations of what that looks like (which might need some digging into)?

To make this point concrete: I would want you to be able to use this to go after the Board of Directors for (hypothetically of course) not being transparent enough. After all, it says "Ensuring transparency" in there. But some people will see that and think "a short, sanitised summary of the minutes every quarter is transparency" while others will expect something comparable to [GitLab's transparency policy](https://about.gitlab.com/handbook/security/transparency_by_default.html). Absent clarity, both are right to understand it the way that they're familiar with. Both of these people might heartily agree with the principles — but when it comes time for either of them to complain about the BoD, neither will have a principle to back them because "transparency" isn't fleshed out enough and we know that it means different things to different people. Do we _really_ disagree that this is the wrong outcome? Wouldn't you rather have a principled stick to beat the Board up with (in this hypothetical, of course)? Or even better, to not have to beat the Board up in the first place because the principle is clear from the get-go?

Regarding hostility, I hear where you are coming from. I sincerely didn't mean providing a doc as "here's a thing, replace yours with it"; I only intended to provide an example for something which I've been trying to convey in different ways. When I heard you push back that the approach is too solidified to change, I concluded that working through an alternative process would be the only way to resolve this. I would much rather work with you and the AB. I'll admit that I am frustrated at not succeeding to convey the issue, though as I said this may well be on me.

You state that "I will note that it is highly unlikely that the entire Membership will agree with everything in the Vision." That fully matches my expectation: what I want to see in a mission/values/vision document is crisp, opinionated statements that people will agree or disagree with. I would be a _lot_ more comfortable if I _disagreed_ with the Vision. But that's precisely the problem I'm pointing at: I see too little in it that I can agree or disagree with. For the most part, I honestly don't know what might be meant by the text because I know those terms have multiple meanings in our community. Consider just this: I think we would all agree that the EWP Sustainability principle is way too short if we were to have substantial discussions of sustainability, but on its own it is 60% of the size of your entire "Principles and Values" section which supposedly covers way more ground. I don't believe that it's possible to be that brief yet crisp and clear. I certainly agree very strongly with: "I would even go so far as to say if the Vision doesn't garner some dissent, it is probably far too weak and not forward-thinking enough." But I don't think that it is currently setting itself up for success on those terms. _Everyone_ agrees that transparency, fairness, privacy, etc. are good things. The dissent is in the details.

This points to an issue that I can only guess at from the outside and am mentioning in case it can help make progress. In this discussion and several side discussions, it seems that AB members past and present believe that the document as it stands is providing defined principles. But as far as I can tell — and I've read it a few times — it's listing the names of principles but not providing much if any indication as to how to understand them. This would indicate one of three things:
1. (Unlikely) The AB believes that these terms admit of a common, uncontroversial definition.
2. The AB is listing terms to scope things out and will define them later. What worries me in this scenario is that I don't see what people are agreeing to: we can both list "fairness" as a goal but we are only agreeing if we agree about what fairness is. (One of several famously contentious terms.)
3. The AB has developed internal consensus about what the terms mean through a few years of discussion, but that has not been transcribed into the document. Based on some comments (eg. "we always intended this to map to HR") I suspect that this is more the case. If so, please understand that this makes it very hard for anyone on the outside to provide any useful contribution until such a time as what is in the AB's minds is written down.

I think that my use of the term "enforcement" may be leading to more disagreement than there is because it isn't (yet) a term of art in our community. @dwsinger mentions using the principles for chartering decisions, grounding appeals and FOs, or resolving FOs. Presumably that means that charters would have a greater chance of being accepted if they align with principles than if they don't, FOs or resolutions if they call upon the principles, etc. That's… enforcement. "If you abide by $x$ then we will make sure that you are more likely to get the outcome you want than if you don't" is pretty much the definition of enforcement: it means the rule is in actual use and has real-world impact.

But for that to work, you need to have principles that give enough material to argue from — it can't just be the name of the principle, especially if it's a principle which we all know is not inherently consensual. If I have a charter that I think (say) "ensures equity" and you object to it on the grounds that you think it doesn't, all we have to go on is the word "equity". It's not that the doc should answer all the questions (as if it could) but it needs to be opinionated if it's going to support arguing. (Once you start defining things more I don't think that it's possible to dodge the question of the relationship to HR — but we can kick that can down the road for now.)

Two thoughts that might help:
* Is there a concrete example of a substantial disagreement in the community over the past few years that the doc as crafted would have helped solved (for instance along the lines that David is thinking about)? I've been involved in a few of those and I can't think of an instance in which I would have benefited from referencing the doc. But I of course haven't been in all of those — maybe the AB is trying to solve a problem that I haven't seen?
* I'm not saying that things should be written defensively, but a way to test whether a principle works would be to kick it adversarially: if Bad Faith Individual (substitute your preferred one) were to try to throw a wrench in our work, how would this help me argue the case with minimal energy? Note that in those kinds of adversarial cases, if it looks like you're making the definition up as you go along it's arguably worse than not having a principle in the first place.

Regarding #17 I understand that it hasn't been rejected, but to repeat what it says in https://github.com/WebStandardsFuture/Vision/pull/37 (which I only now notice you didn't copy over to #17 when you moved — maybe that's part of the misalignment?):

> Dear AB friends, this PR seeks to do two things: 1) to experiment with a longer format (…) Regarding the longer format, as discussed previously, I feel that the vision's principles need to be defined in somewhat greater detail than they currently are if they are to do some work and not just remain a list of nice thoughts that people working to build Web technology soon forget about. Ideas that are expressed too briefly, no matter how good they may be, are often difficult to put into practice. I believe fleshing things out more is more helpful when it comes to support discussions that happen in the trenches. I am eager to hear what your take on this is.

Without alignment on this, I'm not sure what the rest of the PR would do. If there's agreement on this point, maybe then #17 it maps to some combination of "good for people" and "don't centralize" and can then be removed, or maybe it still needs to exist — that all depends on how those principles are defined.

A few further notes on @dwsinger's points:
> The second aspect that puzzles me is the idea that somehow the order – first, write down your principles, and then consider what procedural changes might be needed so that they become embodied in work – is wrong, and we should do it the other way round. I can't imagine getting a good reaction if we were to introduce provisions designed to enforce values and principles, which were to be defined later.

That's not my point. My point is that the principles are only defined to the extent that they are enforced (and vice versa). Let's say we come up with a fictitious principle that our specs must have ipseity. If there is no point at which that principle can be used to impact spec lifecycle (eg. FOs), then people will just ignore it irrespective of the definition. If the Council says that it will rely on it to decide FOs but we don't define it beyond referencing the word "ipseity" then nothing will happen: you can't enforce undefined terms (it's autocracy when you do). If we come up with a lovely detailed definition but the only thing that the Council (or any other enforcement point) uses is that the text should be fuchsia, then people will act as if ipseity=fuchsia.

> it seems to be way early in the process to be making threats (e.g. "Barring meaningful progress on this issue I don't anticipate to be able to vote in favour of the resulting Statement")

This isn't a threat. We were discussing closing the issue, at that point one is supposed to know whether the commenters are satisfied with the answer. (We no longer ask people formally, but I believe it remains a good practice.) If this document were progressed now (or in a generally similar state) I wouldn't vote in favour, I would either abstain or object. I'm leaning abstain because if there's no consensus to make it stronger then I would rather move it out the door and switch to another opportunity to build a stronger statement than get stuck in FO processing. Either way, I don't think that it's fair to interpret a statement of dissent as a threat — it's usually helpful to know where people stand!

**Comment by dwsinger** [comment]
So, if I were to summarize that it's not that the Vision says anything specifically wrong, it's that it needs to be more usable as a bright-line test and/or more actionable, would that capture it?

I think it would be great if we can improve in that direction. I would welcome specific edits that amplify the values and make them more actionable. I worry that the text that makes things actionable would bulk up the document in ways that an external reader might find unhelpful; but structuring or splitting the document (as we did once before with the History section) is easy, once we have text.

**Comment by darobin** [comment]
Taking some time to ponder this, I wonder if we're not disagreeing because we have different expectations of what this document is supposed to do. I can try to break this down into three incremental levels:

1. **A communication frame**. Basically a document that explains at a very high and very generic level what it is that we're doing here. The use of such a document is for comms, for feel-good quotes, for marketing, for recruitment (of people or members). An example might be [The New York Times Mission & Values](https://www.nytco.com/company/mission-and-values/). One way to think about this level is that it documents our charitable purpose more clearly than it has been before, which is definitely useful.
2. **A recital for the W3C**. (We don't have to love the term, it's just descriptive.) In legal documents, a recital typically provides the purpose of the act and some facts about it. It's common in EU laws. A recital isn't normative but it provides background that can be used to discuss and motivate interpretations. (Eg. "We don't have a rule excluding workshops about eating kittens, but the Priority of Constituencies clearly puts cats ahead of all other considerations.") Some constitutions operate at this level.
3. **A foundation for governance of the Web**. This does what a recital does but also establishes constitutional rules for legitimacy, rule-setting and rule-changing, enforceable prescriptions, etc.

If the goal is to produce `1`, then I think it's pretty close to done. I might file some nitpicks, but I would encourage the AB to ship it ASAP. It's a useful thing to have, but I would like to suggest that there are more worrying fires to tend to.

What worries me is that I keep hearing people say that they expect it to work as `2`, and perhaps even as `3`. I strongly disagree that that is the case.

Let's take a simple and obvious example that I would expect a document operating at the `2` level to deal with handily.

---

I am an advertising technologist. I care deeply about keeping the Web free and open, and I'm passionate about standards. I know that a lot of people don't like advertising, but I actually chose this path because I see it as providing essential infrastructure for democracy. I can tell that the whole cookies thing is a mess — you don't need to tell me, I'm the one who has to handle CMP strings and cookie syncing code. My company is honest and we respect user choice, but I wasn't born yesterday and I know that a lot of the rest of the industry isn't quite so honest — though it's always hard to prove. We can do better!

I come to the W3C with a very simple proposal: we should do on the Web what app platforms have done and create the browser equivalent of Apple & Android advertising IDs (so-called "MAIDs" but I don't like that term). I write a simple spec that exposes an efficient, user-controlled unique identifier over an HTTP header and `navigator` field.

I've long known about the W3C but I've never participated, and I've heard it said that people there can be a bit idiosyncratic. I prepare ahead of time by reading the vision document. I'm glad to see that my proposal is fully aligned with the W3C's values! To wit:

* This is for all humanity (it supports a free and independent media) and the good of its users (who don't have to pay).
* It's objectively _a lot_ safer than third-party cookies. In fact, it adds traceability to the system in a way that will make it a lot easier to detect dishonest players.
* It's interoperable, also more than cookies.
* It puts users first: they can always switch the identifier off. This too makes it an improvement over the way in which cookies are supported. It also helps the other constituencies beyond users.
* It is great for privacy: people have full control over it, they can reset or opt out, hide it from some sites, etc. I mean, if both Apple and Google do it, it _must_ be privacy-friendly given what they promise their users.
* All of this helps keep the system trustworthy, the traceability also makes it easier for researchers to find problems.
* It reduces centralisation, you don't get a big player hoarding an identity graph.

With that in mind, I really look forward to working with my new W3C friends!

---

There is nothing contrived about this example, in fact I know quite a few people who match that description, who might make that proposal, and who would reach those conclusions after reading the current Vision doc. Just citing from a talk I attended last night:

!["Behavioural targeting is not only good for consumers (…) it's a rare win for everyone. (…) It ensures that ad placements display content that you might be interested in rather than ads that are irrelevant and uninteresting. (…) Advertisers achieve (…) a greater chance of selling the product. Publishers also win as (…) behavioral targeting increases the value of the ad placements." David Nelson, Operations & IT Director, Unanimis.co.uk](https://user-images.githubusercontent.com/38491/222579888-90a4e9da-413e-4492-871f-c5488000c240.png)

(If you think that kind of statement is made in bad faith or only by bad people, you're wrong.)

There is nothing in the current Vision that can be called upon to reject the above position in a charter or workshop or Council decision that can't also also be used to support it. If you think there is, you're operating on implicit definitions rather than on what the document actually says.

Now, I am confident that I can find copious (but scattered) evidence that such a proposal does not match the values of the W3C or the expectations of the Web community. I know that there's a trove of research out there showing that those motivations don't hold up. But whether this Vision doc exists or not makes no difference to my ability to argue that point.

This is perfectly fine if the goal is `1`. But I find it hard to see how a document that can't support, even a little bit, arguing against the example position above would qualify for `2`. (And it's missing way more for `3`.)

@dwsinger says:
> So, if I were to summarize that it's not that the Vision says anything specifically wrong, it's that it needs to be more usable as a bright-line test and/or more actionable, would that capture it?

 That entirely depends on what it's _for_. To work for `1` I would say it's pretty close. To work for `2` it needs a lot more than a bright line — it needs to at a strict minimum define its salient terms (in the way that the EWP does). That's a much longer document.

**Comment by dennis-dingwei** [comment]
I like the [The New York Times Mission & Values](https://www.nytco.com/company/mission-and-values/) presented by @darobin as the sample for W3C Mission and Values, since which are quite simple and clear, as it really deserves. .
We may need another  Vision document for Web, as the technology W3C is leading.

**Comment by michaelchampion** [comment]
I think resolving #13 would help address the scenario @darobin sketches out above.  i
I proposed some straw-man text in that issue that probably isn't strong enough for Robin's taste, but might help us move us toward drafting text in the   "Vision that can be called upon to reject the above position in a charter or workshop or Council decision" .  Obviously more detailed and authoritative Process or TAG documents would be needed to unambiguously guide such decisions, but it seems reasonable for the Vision to state the general principles behind privacy in more detail.

**Comment by dwsinger** [comment]
Hi Robin

I agree, the document has become overloaded with different purposes, and being clearer about what it is and isn't intended to be will much it much easier to manage. As you say, some of the purposes are in almost direct conflict: an explanation of why we matter, for the outside world, should be written in simple terms and not reference our practices and processes, for example.

I think you have probably enumerated these, but I'll try in different terms to see if we align:
1. A bright light shining into the future that describes in brief, broad, layman's terms, why we matter and where we're going. I think it can be more than feel-good (and should be), strong enough that it's clear what it's _not_ emphasizing.
2. A broad, top-level, 'compass' for internal work, as a basis for more detailed documents, work, processes, and yes, possibly directly decisions (e.g. an FO ruling that prefers a decision that aligns with (1)). Certainly having such documents not align with (1) would be a mistake.
3. A description of how we operate and treat our own constituents. These are also important values, and they are an important part of explaining why people should join (e.g. "you'll be heard"), and why supporting us might also be helpful.
4. More detailed descriptions of consequences, worked-through processes and principles for specific areas; I think that the TAG documents go some way in this direction, more in the direction of worked-through consequences of values than procedural embodiments. These document, in contrast to (1), should use terms of art, should link to and discuss specific technical questions, and so on.

Personally, I'd be happy if it did (1) and (2) (because they should be the same principles, the same document); went light on (3) (because the audience and use is different), and explicitly expect (4) to be done in work in the appropriate venue (TAG, Process CG, PWECG, and so on). I'd also be fine if (2) were a worked-expansion of (1), to make it more of an actionable recital (your word).

I don't think we'll make much progress on 2-4 without 1, though; with only 'full potential' and no Director to embody values, we're almost a de-magnetized magnetic compass. 

If there are top-level compass-bearing types of principles that the Vision doesn't state clearly enough, this is a good time to get to them.

**Comment by cwilso** [comment]
To reply to David's terminology, I agree that we were absolutely starting out as #1, with the understand that this had to grow into/as the basis for other documents that would put this directly into tactical practice (#2).  I believe that we were trying to cover a fair bit of what might be considered #3 as well - notably, we were identifying things that we believe are important in how the W3C focuses inwardly - e.g. community consensus-building, inclusion across a diverse global base, focus on interop - but also with the understanding that what we define as important (e.g. privacy, or sustainability) might change who should support/join.  Happy to excise that from the mission.
#4 sounds like Guides and Process changes, possibly Bylaws.  It is certainly not this document's intent.

To use Robin's terminology, this document should absolutely function as #1 (a communication frame) - and yes, all we really had before was the tagline "Leading the Web to Its Full Potential".
Additionally, I believe it is a definite goal to fill role of #2 (A recital, though no, I don't love the term - I'd suggest "narrative" or "scope" or "ecosystem").  I would ABSOLUTELY AGREE that it does not yet give anywhere near the level of guidance it needs to in order to be complete in this goal; however, as I've said repeatedly, getting some baseline agreement on the 10000ft level prior to working out every level of detail - because even your contrived example has a lot of deep background.
I do not believe this document is the "foundation of governance" for the web (#3); certainly not in this incarnation, and not without being adopted at some point in the future into Bylaws and Process.  But we have literally had nearly nothing actually adopted, even the NYT Mission&Values level, in the past; it's all been "shared agreement" that is generally, well, not necessarily shared when you start pulling at the strings.

**Comment by cwilso** [comment]
I will note that this should likely be resolved as a duplicate of #22 .

**Comment by TzviyaSiegman** [comment]
agreed to close on 10/26/23. If there are narrow topics to continue to discuss, please open new issues
